shared/actions/chat2/index.tsx

```diff
@@ -2353,7 +2362,32 @@ const markThreadAsRead = async (
   if (meta) {
     readMsgID = message ? (message.id > meta.maxMsgID ? message.id : meta.maxMsgID) : meta.maxMsgID
   }
-  logger.info(`marking read messages ${conversationIDKey} ${readMsgID}`)
+
+  logger.info(`marking read messages ${conversationIDKey} ${readMsgID} for ${action.type}`)
+  await RPCChatTypes.localMarkAsReadLocalRpcPromise({
+    conversationID: Types.keyToConversationID(conversationIDKey),
+    msgID: readMsgID,
+  })
+}
+
+const markAsUnread = async (
+  state: Container.TypedState,
+  action: Chat2Gen.MarkAsUnreadPayload,
+  listenerApi: Container.ListenerApi
+) => {
+  if (!state.config.loggedIn) {
+    logger.info('bail on not logged in')
+    return
+  }
+  const conversationIDKey = action.payload.conversationIDKey
+  const readMsgID = action.payload.readMsgID
+  listenerApi.dispatch(
```

Comment:
this doesn't actually need to use the `listernerApi` at all since we can just handle this synchronously. I think you can just do
```
RPCChatTypes.localMarkAsReadLocalRpcPromise({
    conversationID: Types.keyToConversationID(conversationIDKey),
    msgID: readMsgID,
  }).then(()=>{}).catch(()=>{})
return Chat2Gen.createUpdateUnreadline({
      conversationIDKey,
      messageID: readMsgID,
    })
```